docs/cli/ilab-local-config.md

```diff
@@ -0,0 +1,184 @@
+# InstructLab CLI Local Storage Proposal
+
+## Why is this needed?
+
+For tools to be useful to the user, they generally need to work on and know about data pertinent to the user. For example, podman and docker track image blobs, kubectl tracks cluster credentials, etc.
+
+The InstructLab CLI manages a number of settings today such as the teacher model, taxonomy location, inference model, etc. using a local file known as `config.yaml`.
+
+While this method allows ilab to avoid additional complexity of managing a dedicated directory, it’s limited by its ability to scale beyond constraining the user to run ilab in a single directory.
+Since the new ilab CLI will need to not only manage datasets, downloaded models, and any intermediary files, it'll also have to make it simple for the user to orchestrate these things.
+
+We presently store these paths directly in the config, however this
+isn't sustainable as data is frequently moved around unbeknownst to the program. This leaves the user with two choices: either update
+the config to correctly point to either a relative path in the format
+of `../../uncle-dir/dataset.jsonl`, or worse, have to specify an absolute path.
```

Comment:
Why is this worse?

Reply:
It's fragile and very unfriendly to the user.

Reply:
Absolute paths?

Reply:
What happens to your config when you move that file?

Reply:
It breaks, but I don't see how that's worse than a relative path breaking.

Reply:
@nathan-weinberg Think through the motions of using this method. 
First you have to run `ilab init` and make sure your config is in the same directory always. If you're running `ilab` again, you have to either make a new config or navigate back to where that config exists.

Afterwards, you either download a new dataset or generate one from SDG. In either case, you will need to specify what the path to the dataset will be. If you decide to move the dataset or run ilab in another directory, well now you'll need to again either navigate back to where your `config.yaml` exists or have to generate a new one, or specify a `--data-path`. 

Now let's say you decide to rename a file or directory, now you'll need to go back and update all of your config files.
It becomes very tedious to do, and not what we want our users to experience.

Reply:
I agree! I'm definitely not against the approach. It's more the wording of "absolute paths are worse than relative paths when it comes to breaking things" - honestly just removing that language would be fine with me.

Reply:
Don't think this made it into the latest changeset